tjreports/administrator/views/reports/view.html.php

```diff
@@ -46,19 +46,38 @@ class TjreportsViewReports extends JViewLegacy
 	 */
 	public function display($tpl = null)
 	{
-		$canDo = TjreportsHelpersTjreports::getActions();
-		$user       = JFactory::getUser();
-		$user_id    = $user->id;
-		$input = JFactory::getApplication()->input;
+		$extension = JFactory::getApplication()->input->get('extension', '', 'STRING');
+
+		$full_client = explode('.', $extension);
 
-		if (!$canDo->get('view.reports'))
+		// Eg com_tjlms
+		$component = $full_client[0];
+		$eName = str_replace('com_', '', $component);
+		$file = JPath::clean(JPATH_ADMINISTRATOR . '/components/' . $component . '/helpers/' . $eName . '.php');
+
+		if (file_exists($file))
 		{
-			JError::raiseError(500, JText::_('JERROR_ALERTNOAUTHOR'));
+			require_once $file;
+
+			$prefix = ucfirst(str_replace('com_', '', $component));
+			$cName = $prefix . 'Helper';
+
+			if (class_exists($cName))
+			{
+				$canDo = TjlmsHelper::getActions();
```

Comment:
Is this correct ?  Where are you using $cName ?

Also, can you explain a little what is happening here ?